adoc/headers/vec.h

```diff
@@ -121,112 +121,65 @@ template <typename DataT, int NumElements> class vec {
+  friend vec& operatorOP(vec& lhs, const DataT& rhs);
 
   // OP is: &&, ||
-  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs) {
-    /* ... */ }
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs) {
-    /* ... */ }
-
-    // OP is: <<, >>
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const vec& lhs, const vec& rhs) { /* ... */
-    }
-    friend vec operatorOP(const vec& lhs, const DataT& rhs) { /* ... */
-    }
-
-    // OP is: <<=, >>=
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec& operatorOP(vec& lhs, const vec& rhs) { /* ... */
-    }
-    friend vec& operatorOP(vec& lhs, const DataT& rhs) { /* ... */
-    }
-
-    // OP is: ==, !=, <, >, <=, >=
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs) {
-    /* ... */ }
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs) {
-    /* ... */ }
-
-    vec& operator=(const vec<DataT, NumElements>& rhs);
-    vec& operator=(const DataT& rhs);
-
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operator~(const vec& v) { /* ... */
-    }
-    friend vec<RET, NumElements> operator!(const vec& v) { /* ... */
-    }
-
-    // OP is: +, -, *, /, %
-    /* operator% is only available when: DataT != float && DataT != double &&
-    DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: &, |, ^
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: &&, ||
-    friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs) {
-    /* ... */ }
-
-    // OP is: <<, >>
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: ==, !=, <, >, <=, >=
-    friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs) {
-    /* ... */ }
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs);
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs);
+  friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs);
+
+  // OP is: ==, !=, <, >, <=, >=
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs);
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs);
+  friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs);
+
+  vec& operator=(const vec<DataT, NumElements>& rhs);
+  vec& operator=(const DataT& rhs);
+
+  /* Available only when: DataT != float && DataT != double && DataT != half. */
+  friend vec operator~(const vec& v);
+  friend vec<RET, NumElements> operator!(const vec& v);
```

Comment:
Can you put a blank link between these two?  This will help make it clear that the constraint ("Available only when ...") applies only to `operator~`.  The table already makes this clear too.